pallets/multisig/src/lib.rs

```diff
@@ -733,6 +737,11 @@ impl<T: Trait> Module<T> {
             Error::<T>::AlreadyApproved
         );
         if let Some(proposal) = Self::proposals(&multisig_proposal) {
+            // Check whether the proposal is already closed or not.
+            ensure!(
+                !Self::is_proposal_closed(&multisig_proposal),
```

Comment:
I think this is OK - the alternative approach would be to still count votes / approvals, and just not execute (i.e. move these lines to below `<TxApprovals<T>>::insert(&multisig_proposal, approvals);` - this may be easier to debug later if we see the full voting history. We could also consider not returning an error, but just exiting without executing the proposal.

@maxsam4 for his thoughts.

Reply:
Overall, I think it is a good idea to store all approvals but execute only once. I can see it coming handy while debugging.

If we store the approval but do not execute then we should return success instead of an error.

Reply:
Makes sense. @satyamakgec please can we make this minor changes.

Reply:
The only reason I choose this approach because it is possible to change the no. of signature required for approval then it may open a window to re-execute a proposal. But to avail above discussed approach, I am keeping the `is_proposal_closed()`  functionality, and instead of spitting an error It will return `Ok(())`  and `ProposalApproved` Event.

Reply:
Great thanks - I think that even if the threshold changes, it won't be possible to re-execute a proposal with the current code (as it will be marked closed the first time it's executed in any case).